src/main/webapp/app/entities/oauth-client/oauth-client-dialog.component.ts

```diff
@@ -0,0 +1,186 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Response } from '@angular/http';
+
+import { NgbActiveModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { EventManager, AlertService, JhiLanguageService } from 'ng-jhipster';
+
+import { OAuthClient } from './oauth-client.model';
+import { OAuthClientService } from './oauth-client.service';
+import { OAuthClientPopupService } from './oauth-client-popup.service';
+
+@Component({
+    selector: 'jhi-oauth-client-dialog',
+    templateUrl: './oauth-client-dialog.component.html'
+})
+export class OAuthClientDialogComponent implements OnInit  {
+
+    client: OAuthClient;
+    isSaving: boolean;
+    authorities: any[];
+    grantTypes: any[];
+    scopeList: string;
+    resourcesList: string;
+    autoApproveScopeList: string;
+    availableGrants: string[];
+    dynamicRegistration: boolean;
+    dynamicRegistrationKey: string;
+    newClient: boolean;
+    protectedClient: boolean;
+
+    constructor(
+        public activeModal: NgbActiveModal,
+        private jhiLanguageService: JhiLanguageService,
+        private alertService: AlertService,
+        private oauthClientSerivce: OAuthClientService,
+        private eventManager: EventManager
+    ) {
+        this.jhiLanguageService.setLocations(['oauthclient']);
```

Comment:
should be `this.jhiLanguageService.setLocations(['oauthClient']);`

Reply:
note that here and in other places: a dialog should generally use `addLocation`, a page could use `setLocation`. This prevents other translations on the page to be lost.